Assets/MixedRealityToolkit/_Core/Interfaces/Sharing/ISimpleState.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Threading.Tasks;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Interfaces.Sharing
+{
+    /// <summary>
+    /// Simplest possible non-automatic state sharing.
+    /// Will typically be networked but could also plug into a file system or local database.
+    /// </summary>
+    public interface ISimpleState : IMixedRealityService
+    {
+        Task<bool> HasKey(string key);
+        Task SetState<T>(string key, T state);
+        Task<T> GetState<T>(string key);
```

Comment:
It seems like this GetState may also fail/we may want to think about defining this as some sort of Task<bool, T> compared to only a Task<T>

Reply:
Maybe we could let devs pass an optional default that gets returned in the event of a failure?
`Task<T> GetState<T>(string key, T default);`
Or it could just throw an exception. For this system simplicity should take priority over robustness.

Reply:
Asynchronous/multi-threaded exception handling seems a bit more difficult than assessing a return value. I'd suggest the default value or a valueOnFailure argument compared to throwing.